test/tstCallbacks.cpp

```diff
@@ -105,4 +138,33 @@ int main()
 
   // check_valid_callback(ArborX::Details::CallbackDefaultNearestPredicate{},
   //                     SpatialPredicates{}, v);
+
+  check_valid_callback(CustomCallbackNearestPredicate{}, NearestPredicates{});
+
+  check_valid_callback(CustomCallbackSpatialPredicate{}, SpatialPredicates{});
+
+  check_valid_callback(CustomCallbackSpatialPredicateMissingNonVoidReturnType{},
+                       SpatialPredicates{});
+
+  // Uncomment to see error messages
+
+  // check_valid_callback(CustomCallbackSpatialPredicateMissingConstQualifier{},
+  //                     SpatialPredicates{});
+
+  // check_valid_callback(CustomCallbackNearestPredicate{},
+  // SpatialPredicates{});
+
+  // check_valid_callback(CustomCallbackSpatialPredicate{},
+  // NearestPredicates{});
+
+#ifndef __NVCC__
+  check_valid_callback([](auto const & /*predicate*/, int /*primitive*/) {},
+                       SpatialPredicates{});
+
+  check_valid_callback(
+      [](auto const & /*predicate*/, int /*primitive*/, float /*distance*/) {},
+      NearestPredicates{});
+
+  // check_valid_callback([](Wrong, int /*primitive*/) {}, SpatialPredicates{});
```

Comment:
Is this to be uncommented to see an error?

Reply:
Yep

Reply:
Please add a comment to that.

Reply:
Did new commit fixup, rebased, and pushed force